packages/components/src/styles/components/form/layout.scss

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// FORM LAYOUT COMPONENTS
+//
+
+// FORM
+
+// Default max-width for form sections & similar components
+$hds-form-section-max-width: 768px;
+
+.hds-form {
+  --hds-form-section-max-width: #{$hds-form-section-max-width};
+
+  display: flex;
+  flex-direction: column;
+  gap: 32px;
+
+  // Set a max-width for the ButtonSet when it's within an Hds::Form
+  // Q: Should this be done (max-width on ButtonSet) and if so should it be done here or in the ButtonSet component?
+  .hds-button-set {
+    max-width: var(--hds-form-section-max-width, #{$hds-form-section-max-width});
+  }
+}
+
+// FORM HEADER & SECTION HEADER
+
+.hds-form__header,
+.hds-form__section-header {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+}
+
+// Note: Only the Form Header needs a max-width as the Section Header is used within a Section
```

Comment:
[thought] not necessarily true: as a consumer I may use the `<Hds::Form::Section::Header>` without a wrapping `<Hds::Form::Section>` and in terms of vertical spacing it would work perfectly, and I may not notice that the width is not limited. if the intention is that the SectionHeader has always a max-width, I would make it explicit in the CSS

Reply:
The `<Hds::Form::Section::Header>` is meant to be nested inside a `<Hds::Form::Section>` as I conceived it. I don't think the spacing would be quite right if it weren't nested either, although I can double-check that.

I mentioned this behavior to @LilithJames-HDS previously btw. (About the SectionHeader using the same width as its Section)